Telegraf v1.33.0 (#5708)
influxdata/docs-v2: content/telegraf/v1/release-notes.md

- [#15742](https://github.com/influxdata/telegraf/pull/15742) `outputs.influxdb_v2` Add rate limit implementation
- [#15943](https://github.com/influxdata/telegraf/pull/15943) `outputs.mqtt` Add sprig functions for topic name generator
- [#16041](https://github.com/influxdata/telegraf/pull/16041) `outputs.postgresql` Allow limiting of column name length
- [#16258](https://github.com/influxdata/telegraf/pull/16258) `outputs` Add rate-limiting infrastructure
- [#16146](https://github.com/influxdata/telegraf/pull/16146) `outputs` Implement partial write errors
- [#15883](https://github.com/influxdata/telegraf/pull/15883) `outputs` Only copy metric if its not filtered out
- [#15893](https://github.com/influxdata/telegraf/pull/15893) `serializers.prometheusremotewrite` Log metric conversion errors

### Bugfixes

- [#16248](https://github.com/influxdata/telegraf/pull/16248) `inputs.netflow` Decode flags in TCP and IP headers correctly
- [#16257](https://github.com/influxdata/telegraf/pull/16257) `inputs.procstat` Handle running processes correctly across multiple filters

> - [#16219](https://github.com/influxdata/telegraf/pull/16219) `logging` Add Close() func for redirectLogger

Reviewer: ```suggestion
- [#16219](https://github.com/influxdata/telegraf/pull/16219) `logging` Add `Close()` func for redirectLogger
```

- [#16201](https://github.com/influxdata/telegraf/pull/16201) `deps` Bump github.com/intel/powertelemetry from 1.0.1 to 1.0.2
- [#16200](https://github.com/influxdata/telegraf/pull/16200) `deps` Bump github.com/rclone/rclone from 1.68.1 to 1.68.2
- [#16199](https://github.com/influxdata/telegraf/pull/16199) `deps` Bump github.com/vishvananda/netns from 0.0.4 to 0.0.5
- [#16236](https://github.com/influxdata/telegraf/pull/16236) `deps` Bump golang.org/x/net from 0.30.0 to 0.31.0
- [#16250](https://github.com/influxdata/telegraf/pull/16250) `deps` Bump golangci-lint from v1.62.0 to v1.62.2
- [#16233](https://github.com/influxdata/telegraf/pull/16233) `deps` Bump google.golang.org/grpc from 1.67.1 to 1.68.0
- [#16202](https://github.com/influxdata/telegraf/pull/16202) `deps` Bump modernc.org/sqlite from 1.33.1 to 1.34.1
- [#16203](https://github.com/influxdata/telegraf/pull/16203) `deps` Bump super-linter/super-linter from 7.1.0 to 7.2.0

## v1.32.3 {date="2024-11-18"}

### Important Changes

- PR [#16015](https://github.com/influxdata/telegraf/pull/16015) changes the

>   internal counters of the Bind plugin to unsigned integers matching the server
>   implementation. We keep backward compatibility by setting

Reviewer: ```suggestion
  internal counters of the Bind plugin to unsigned integers (as in the server
  implementation). For backward compatibility,
```

>   `report_counters_as_int` to `true` by default to avoid type conflicts on the

Reviewer: ```suggestion
  `report_counters_as_int` is `true` by default to avoid type conflicts on the
```

>   output side. However, you should change this setting to `false` as soon as
>   possible to avoid invalid values and parsing errors with the v3 XML
>   statistics.

Reviewer: ```suggestion
  output side. _However, you should set  `report_counters_as_int` to `false` as soon as
  possible to avoid invalid values and parsing errors with the v3 XML
  statistics._
```

### Bugfixes

> - [#16123](https://github.com/influxdata/telegraf/pull/16123) `agent` Restore setup order of stateful plugins to Init() then SetState()

Reviewer: ```suggestion
- [#16123](https://github.com/influxdata/telegraf/pull/16123) `agent` Restore setup order of stateful plugins to `Init()` then `SetState()`
```

  delivery state update of un-parseable messages from `ACK` to `NACK` without
  requeueing. This way, those messages are not lost and can optionally be
  handled using a dead-letter exchange by other means.

> - Removal of old-style serializer creation. This should not directly affect
>   users as it is an API change. All serializers in Telegraf are already ported
>   to the new framework. If you experience any issues with not being able to
>   create serializers let us know!

Reviewer: ```suggestion
- Removal of old-style serializer creation. This should not directly affect
  users as it is an API change; all serializers in Telegraf are already ported
  to the new framework. If you experience any issues creating serializers, [contact us](https://docs.influxdata.com/telegraf/v1/#bug-reports-and-feedback).
```
